Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -140,14 +141,8 @@ export class azureclitask {
     }
 
     private static setConfigDirectory(): void {
-        var configDirName: string = "c" + new Date().getTime(); // 'c' denotes config
-        if (tl.osType().match(/^Win/)) {
-            this.azCliConfigPath = path.join(process.env.USERPROFILE, ".azclitask", configDirName);
-        }
-        else {
-            this.azCliConfigPath = path.join(process.env.HOME, ".azclitask", configDirName);
-        }
-
+        var configDirName: string = "config" + new Date().getTime();
+        this.azCliConfigPath = path.join(tl.getVariable('Agent.TempDirectory'), ".azclitask", configDirName);
```

Comment:
.azclitask can be removed from directory hierarchy. It was useful when we were creating config dir in a user visible location. Agent temp is much less intrusive and hence no need to have .azclitask dir.